com.unity.multiplayer.mlapi/Tests/Runtime/InterestTests.cs

```diff
@@ -129,6 +130,58 @@ public override void UpdateObject(in NetworkObject obj)
             return (no, objGuid);
         }
 
+        [Test]
+        // Start is called before the first frame update
+        public void InterestRandomTest()
+        {
+            InterestNodeStatic randomNode = ScriptableObject.CreateInstance<InterestNodeStatic>();
+            var randomKernel = ScriptableObject.CreateInstance<RandomInterestKernel>();
+            randomNode.InterestKernels.Add(randomKernel);
+            var staticNode = ScriptableObject.CreateInstance<InterestNodeStatic>();
+
+            var results = new HashSet<NetworkObject>();
+            var nc = new NetworkClient()
+            {
+                ClientId = 1,
+            };
+
+            var (playerObj, playerGuid) = MakeGameInterestObjectHelper(new Vector3(0.0f, 0.0f, 0.0f), null);
+            nc.PlayerObject = playerObj;
+
+            var (random1Obj, random1) = MakeGameInterestObjectHelper(new Vector3(0.5f, 0.0f, 0.0f), randomNode);
+            NetworkManagerHelper.SpawnNetworkObject(random1);
+
+            var (random2Obj, random2) = MakeGameInterestObjectHelper(new Vector3(1.0f, 0.0f, 0.0f), randomNode);
+            NetworkManagerHelper.SpawnNetworkObject(random2);
+
+            var (random3Obj, random3) = MakeGameInterestObjectHelper(new Vector3(3.0f, 0.0f, 0.0f), randomNode);
+            NetworkManagerHelper.SpawnNetworkObject(random3);
+
+            var (alwaysObj, alwaysGuid)  = MakeGameInterestObjectHelper(new Vector3(99.0f, 99.0f, 99.0f), staticNode);
+            NetworkManagerHelper.SpawnNetworkObject(alwaysGuid);
+
+            NetworkManagerHelper.SpawnNetworkObject(playerGuid);
+
+            results.Clear();
+            NetworkManager.Singleton.InterestManager.QueryFor(nc, results);
+
+            Assert.True(!randomKernel.VisibleObjects.Contains(random1Obj) || results.Contains(random1Obj));
```

Comment:
Ok, if I read this right, this makes sure that the kernel gets invoked and therefore VisibleObjects needs to match the results. 

But if we're trying to test the random aspect, this could be stronger.  What I would probably do is:

1. Add a "set seed" method to the RandomInterestKernel
2. Then in your test, call Random.Value n times (where n is substantially more than 3).  Now you know how many to expect to come through
3. Now call setSeed again, and run the tests, and make sure you get the same number of hits
4. I think with this you can also eliminate the need for the VisibleObjects tracker

Reply:
Took a slightly different route on the test coverage, but hopefully end up with the same or better functional validation